aml/src/pci_routing.rs

```diff
@@ -174,7 +178,7 @@ impl PciRoutingTable {
                 let resources = resource::resource_descriptor_list(link_crs)?;
                 match resources.as_slice() {
                     [Resource::Irq(descriptor)] => Ok(descriptor.clone()),
-                    _ => Err(AmlError::IncompatibleValueConversion),
+                    _ => Err(AmlError::IncompatibleValueConversion { current: todo!(), target: todo!() }),
```

Comment:
Hmm, this probably shouldn't even be emitting `IncompatibleValueConversion`, which is why there aren't obvious types to put here. Perhaps we should add a new error type (e.g. `UnexpectedResourceType` in `AmlError` (next to the other resource error variants)) and emit that here.

Reply:
I have added `AmlError::UnexpectedResourceType` to handle this case.